### PR TITLE
Trivial: Minor header cleanup

### PR DESCRIPTION
**Pull request by bstoeger**
### Describe the pull request:

- [ ] Bug fix
- [ ] Functional change
- [ ] New feature
- [x] Code cleanup
- [ ] Build system change
- [ ] Documentation change
- [ ] Language translation

### Pull request long description:

Minor header cleanup in `import-csv.h`. Make one local function static. And remove redundant prototypes. These were a bit scary, because in `dive.h` they were declared `extern "C"`, whereas in `import-csv.h` they were not. I have my doubts that this is well defined.
 
### Changes made:

1) Make function static.
2) Remove redundant prototypes from `dive.h`.

### Release note:

No.
### Documentation change:

No.

### CODE REVIEW

**Comment by dirkhh** [comment]
Declarations pulled in from C code behave the same as `extern "C"` ones in C code.
It's C++ code where this changes things.

File: core/import-csv.h [hunk 1]
```diff
@@ -17,11 +17,19 @@ enum csv_format {
 
 #define MAXCOLDIGITS 10
 
+#ifdef __cplusplus
+extern "C" {
+#endif
```
Comment (dirkhh) [inline core/import-csv.h:22]: interesting - we include this in divelogimportdialog.cpp
I wonder why this didn't cause us problems...

Comment (dirkhh) [reply core/import-csv.h:22]: Now I know why. We don't need that include and should remove it.
Which means that technically we don't need that `#ifdef` bracing, but it's not wrong, either.

File: desktop-widgets/mainwindow.cpp [hunk 2]
```diff
@@ -23,6 +23,7 @@
 #include "core/gettextfromc.h"
 #include "core/git-access.h"
 #include "core/isocialnetworkintegration.h"
+#include "core/import-csv.h"
```
Comment (dirkhh) [inline desktop-widgets/mainwindow.cpp:26]: hmm - so why did you add that here? Now I'm confused.

Comment (dirkhh) [reply desktop-widgets/mainwindow.cpp:26]: oh never mind, ignore all of my comments. I need more coffee.

**Comment by dirkhh** [comment]
After thoroughly confusing myself, I now get what this actually does. Approved.

File: core/import-csv.h (hunk 1 above)
Comment (bstoeger) [reply core/import-csv.h:22]: It worked because the extern "C" was taken from dive.h.
